cwms-data-api/src/main/java/cwms/cda/api/watersupply/AccountingCreateController.java

```diff
@@ -0,0 +1,164 @@
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+
+public class AccountingCreateController implements Handler {
+    private static final String TAG = "Pump Accounting";
+    private final MetricRegistry metrics;
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    public AccountingCreateController(MetricRegistry metrics) {
+        this.metrics = metrics;
+    }
+
+    @NotNull
+    protected WaterSupplyAccountingDao getWaterSupplyAccountingDao(DSLContext dsl) {
+        return new WaterSupplyAccountingDao(dsl);
+    }
+
+    @OpenApi(
+        requestBody = @OpenApiRequestBody(
+            content = {
+                @OpenApiContent(from = WaterSupplyAccounting.class, type = Formats.JSONV1)
+            },
+            required = true),
+        pathParams = {
+            @OpenApiParam(name = OFFICE, description = "The office ID the accounting is associated with.",
+                    required = true),
+            @OpenApiParam(name = WATER_USER, description = "The water user the accounting is associated with.",
+                    required = true),
+            @OpenApiParam(name = CONTRACT_NAME, description = "The name of the contract associated with the "
+                    + "accounting.", required = true),
+        },
+        responses = {
+            @OpenApiResponse(status = STATUS_204, description = "The pump accounting entry was created."),
+            @OpenApiResponse(status = STATUS_501, description = "Requested format is not implemented")
+        },
+        description = "Create a new pump accounting entry associated with a water supply contract.",
+        path = "/projects/{office}/water-user/{water-user}/contracts/{contract-name}/accounting",
+        method = HttpMethod.POST,
+        tags = {TAG}
+    )
+
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(CREATE)) {
+            final String contractId = ctx.pathParam(CONTRACT_NAME);
+            final String office = ctx.pathParam(OFFICE);
+            DSLContext dsl = getDslContext(ctx);
+            String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) : Formats.JSONV1;
+            ContentType contentType = Formats.parseHeader(formatHeader, WaterSupplyAccounting.class);
+            ctx.contentType(contentType.toString());
+            WaterSupplyAccounting accounting = Formats.parseContent(contentType, ctx.body(),
+                    WaterSupplyAccounting.class);
+            WaterSupplyAccountingDao waterSupplyAccountingDao = getWaterSupplyAccountingDao(dsl);
+            LookupTypeDao lookupTypeDao = new LookupTypeDao(dsl);
+            List<LookupType> lookupList = lookupTypeDao
```

Comment:
The database doesn't already perform these checks in the store call?

Reply:
I double checked, it seems the database will drop any data points that have a transfer type that does not exist when storing accounting data. No error is thrown, those values are just not stored and do not come back upon retrieval.

Reply:
This should get logged as a ticket in Jira for the database schema. Can you add this as a comment in the CDA code?

Reply:
Added a comment and created an issue ticket: https://jira.hecdev.net/browse/CWDB-316